classes/cmake.yaml

```diff
@@ -71,27 +76,73 @@ buildScript: |
+        if [[ ${CMAKE_GENERATOR} == "Visual Studio"* ]] ; then
+            INSTALL_PREFIX="$DESTDIR$INSTALL_PREFIX"
+            # since VS2019 we can select the platform and target by special variables
+            OPTIONS+=( ${CMAKE_GENERATOR_PLATFORM:+-A"${CMAKE_GENERATOR_PLATFORM}"} )
+            OPTIONS+=( ${CMAKE_GENERATOR_TOOLSET:+-T"${CMAKE_GENERATOR_TOOLSET}"} )
+        else
+            # windows does not support a sigle build type, we have to select the
+            # current type with cmake --build . --config
+            OPTIONS+=( -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE:-RelWithDebInfo} )
+        fi
+
+        # for find_package we use this cmake install path structure:
+        # <prefix>/(lib/<arch>|lib|share)/cmake/<name>*/ (U)
+        # this one is just supported for UNIX. For Windows we can use this:
+        # <prefix>/<name>*/ (W)
+        # by adding the CMAKE_PREFIX_PATH /usr/lib/cmake. For UNIX we add /usr.
         cmake "$1" \
             ${CMAKE_TOOLCHAIN_FILE:+-DCMAKE_TOOLCHAIN_FILE="$CMAKE_TOOLCHAIN_FILE"} \
             -DCMAKE_FIND_ROOT_PATH="$CMAKE_FIND_ROOT_PATH" \
-            -DCMAKE_BUILD_TYPE=Bob \
-            -DCMAKE_INSTALL_PREFIX="/usr" \
-            -DBUILD_SHARED_LIBS=$BUILD_SHARED_LIBS \
+            -DCMAKE_PREFIX_PATH="/usr/lib/cmake;/usr" \
+            -DCMAKE_INSTALL_PREFIX="$INSTALL_PREFIX" \
+            ${BUILD_SHARED_LIBS:+-DBUILD_SHARED_LIBS=$BUILD_SHARED_LIBS} \
+            ${OPTIONS:+"${OPTIONS[@]}"} \
             "${@:2}"
-        makeParallel $MAKE_TARGET ${MAKE_OPTIONS:+"${MAKE_OPTIONS[@]}"}
-        if [[ -n "$INSTALL_TAGET" ]] ; then
-            make $INSTALL_TAGET \
-                DESTDIR=${PWD}/../install \
+
+        if [[ ${CMAKE_GENERATOR} == "Visual Studio"* ]] ; then
+            cmake --build . --parallel "${MAKE_JOBS-$(nproc)}" \
+                --config ${CMAKE_BUILD_TYPE:-RelWithDebInfo} \
+                ${MAKE_TARGET:+--target "${MAKE_TARGET}"} \
                 ${INSTALL_OPTIONS:+"${INSTALL_OPTIONS[@]}"}
+        else
+            makeParallel $MAKE_TARGET ${MAKE_OPTIONS:+"${MAKE_OPTIONS[@]}"}
+        fi
+        if [[ -n "$INSTALL_TARGET" ]] ; then
+            if [[ ${CMAKE_GENERATOR} == "Visual Studio"* ]] ; then
+                if [[ "$INSTALL_TARGET" == "install" ]] ; then
+                    cmake --install . \
+                        --config ${CMAKE_BUILD_TYPE:-RelWithDebInfo} \
+                        ${INSTALL_OPTIONS:+"${INSTALL_OPTIONS[@]}"}
+                else
+                    # will be built again...
+                    cmake --build . --parallel "${MAKE_JOBS-$(nproc)}" \
+                        --config ${CMAKE_BUILD_TYPE:-RelWithDebInfo} \
+                        --target $INSTALL_TARGET \
+                        ${INSTALL_OPTIONS:+"${INSTALL_OPTIONS[@]}"}
+                fi
+            else
+                make $INSTALL_TARGET \
+                    DESTDIR=$DESTDIR \
```

Comment:
I will check, if that will change anything in Linux, if we use cmakes install_prefix instead of the DESTDIR of make.